src/index.js

```diff
@@ -56,29 +59,31 @@ class Libp2p extends EventEmitter {
 
     // Setup the Upgrader
     this.upgrader = new Upgrader({
-      localPeer: this.peerInfo.id,
+      localPeer: this.peerId,
       metrics: this.metrics,
       onConnection: (connection) => {
-        const peerInfo = new PeerInfo(connection.remotePeer)
-        this.registrar.onConnect(peerInfo, connection)
+        const peerId = connection.remotePeer
+
+        this.registrar.onConnect(peerId, connection)
         this.connectionManager.onConnect(connection)
-        this.emit('peer:connect', peerInfo)
+        this.emit('peer:connect', peerId)
```

Comment:
Something to think about since we need to break the API here anyway, should we emit the actual connection here? I think this could be useful. We'll have access to the PeerId via `connection.remotePeer`. It will also make it easy to act on the specific connection of the event, since we can potentially have multiple connections to a peer. Currently, when we get this event there is no way to map this back to a specific connection if we have more than 1 to that peer.

Reply:
yeah, I got that issue on #611 . I will add this comment there and change it there :)

Reply:
https://github.com/libp2p/js-libp2p/pull/611#discussion_r414504819